### PR TITLE
feat: add bazaar auction receipt token

### PR DESCRIPTION
**Pull request by penandlim**
## Describe your changes

Add a token to be used for the bazaar auction

## Checklist before requesting a review

- [ ] Title follows [conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Newly added functions follow Check-effects-interaction
- [ ] Gas usage has been minimized (ex. Storage variable access is minimized)

### CODE REVIEW

File: src/governance/CoveTokenBazaar.sol [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    event ReceiverAllowed(address indexed target, uint256 eventId);
+    /**
+     * @notice Emitted when an address has its permission to receive transfers revoked.
+     * @param target The address that is being disallowed from receiving tokens.
+     * @param eventId An identifier for the event to order events within the same block.
+     */
+    event ReceiverDisallowed(address indexed target, uint256 eventId);
+
+    /**
+     * @notice Deploys this contract with the initial owner and minting allowed after a specified time.
```
Comment (lilPlumberBoy) [inline src/governance/CoveTokenBazaar.sol:56]:
```suggestion
     * @notice Deploys this contract with the initial owner minted the total supply.
```

File: test/unit/governance/CoveTokenBazaarAuction.t.sol [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        coveTokenBazaarAuction.removeAllowedSender(user);
+        assertTrue(!coveTokenBazaarAuction.allowedSender(user), "User should not be allowed to transfer");
+        coveTokenBazaarAuction.transfer(user, amount);
+        vm.expectRevert(Errors.TransferNotAllowedYet.selector);
+        vm.stopPrank();
+        vm.prank(user);
+        coveTokenBazaarAuction.transfer(alice, amount);
+    }
+
+    function testFuzz_removeFromAllowedSender_revertWhen_CallerIsNotTimelock(address user) public {
```
Comment (lilPlumberBoy) [inline test/unit/governance/CoveTokenBazaarAuction.t.sol:166]:
```suggestion
    function testFuzz_removeFromAllowedSender_revertWhen_CallerIsNotOwner(address user) public {
```

File: test/unit/governance/CoveTokenBazaarAuction.t.sol [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        amount = bound(amount, 0, TOTAL_SUPPLY);
+        vm.startPrank(owner);
+        coveTokenBazaarAuction.addAllowedReceiver(user);
+        vm.stopPrank();
+        vm.prank(user);
+        vm.expectRevert(Errors.TransferNotAllowedYet.selector);
+        coveTokenBazaarAuction.transfer(owner, amount);
+    }
+
+    function testFuzz_addAllowedSender_revertWhen_CallerIsNotTimelock(address user) public {
```
Comment (lilPlumberBoy) [inline test/unit/governance/CoveTokenBazaarAuction.t.sol:144]:
```suggestion
    function testFuzz_addAllowedSender_revertWhen_CallerIsNotOwner(address user) public {
```

**Review by lilPlumberBoy — APPROVED** [review]
Small nits, overall LGTM

File: src/governance/CoveTokenBazaarAuction.sol [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+import { Errors } from "src/libraries/Errors.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { Ownable2Step, Ownable } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+
+/**
+ * @title CoveTokenBazaarAuction
+ * @notice ERC20 token with capped supply and owner controlled allowlisting mechanism for transfers.
+ * @dev Intended usage of the allowlisting mechanism is:
+ * - Senders: Bazaar Uncapped Auction Factory for creating the auction, Bazaar Uncapped Auction for users to claim
```
Comment (alphastorm) [inline src/governance/CoveTokenBazaarAuction.sol:12]: nit: also the community multisig

File: src/governance/CoveTokenBazaarAuction.sol [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+     * @notice Deploys this contract with the initial owner and minting the total supply to the owner.
+     * @param owner_ The address of the initial owner.
+     */
+    // slither-disable-next-line locked-ether
+    constructor(address owner_) payable Ownable() ERC20("Cove DAO Bazaar Auction Token", "COVE-BAZAAR") {
+        // Checks
+        if (owner_ == address(0)) {
+            revert Errors.ZeroAddress();
+        }
+        _transferOwnership(owner_); // Transfer ownership to the provided address
```
Comment (alphastorm) [inline src/governance/CoveTokenBazaarAuction.sol:65]: does this mean the community multisig will also have to accept ownership before allowlisting the created bazaar auction contract?

Comment (penandlim) [reply src/governance/CoveTokenBazaarAuction.sol:65]: Nope. Only the external calls are 2 step. Unlikely we will take advantage of the 2step ownership transfers for this token but decided to side with the best practices.

**Pull request closed without merging**